src/Core/Services/TypeHelper.cs

```diff
@@ -151,6 +180,15 @@ public static JsonDataType GetJsonDataTypeFromSystemType(Type type)
         /// <returns>DbType for the given system type. Null when no mapping exists.</returns>
         public static DbType? GetDbTypeFromSystemType(Type systemType)
         {
+            // Get the underlying type argument if the 'systemType' argument is a nullable type.
+            Type? nullableUnderlyingType = Nullable.GetUnderlyingType(systemType);
```

Comment:
A nullable value type like `int? myInt` is represented by `Nullable<T>` -> `Nullable<int>`. So `Type? systemType` wouldn't apply.

`Nullable.GetUnderlyingType(Type systemType)` would resolve `int` from the passed in `int?` /  `Nullable<int>`.

Then the question switches to nullable reference types like `string?`. Per [csharplang discussion](https://github.com/dotnet/csharplang/discussions/3003#discussioncomment-129629), `string?` is represented in runtime as `string` with additional type metadata:
> Because nullable reference type does not have a standalone runtime representation. When using reflection, it's represented by NullableContextAttribute on the target member.

Whether we ever resolve a nullable value/reference type from the database is another question. I ran the type tests locally from 'main' and removed all the nullable value types from `_systemTypeToDbTypeMap`, and didn't come across failures in tests containing the string 'type'. I don't recall why the nullable types were in that collection to begin with. Because they were present and I didn't want to risk regressions in #1568 with regards to type handling, i added the helper replicated here for pulling the underlying type out of `Nullable<T>` /nullable value types. I also had written tests to cover nullable value types if they ever occurred: https://github.com/Azure/data-api-builder/blob/60e38a2a3c2cdb250ad71582becd69d9388ffa73/src/Service.Tests/OpenApiDocumentor/CLRtoJsonValueTypeUnitTests.cs#L108

Reply:
I get it now.